### ISSUE TITLE
New Loss function and Metric to monitor best weights

### ISSUE THREAD

**Issue by herleeyandi** [fchollet/keras#6412]
Hello I am still exploring keras in the past few months so I am new here. I like it so much thanks for @fchollet  . I just have 3 problem now.
1) How we can create new loss function?[better with example]
In here for example I have 4 pairs of coordinate `p = x1,y1,x2,y2,x3,y3,x4,y4` which allocated in 1x8 array as the output of network. I want to create euclidean loos function which compare `y_pred` and `y_true` so I will have 4 distance as the result comparing every point in the array. The problem is I just know that I must use Keras Backend operation. Now the proble is how can I iterate through the tensor? for example I am taking the mean of all 4 distance:
`def euclidean(y_true, y_pred):`
`    ktotal = K.zeros((1,4))`
`    i = 0`
`    j = 0`
`    while(i<7):`
`        ktotal[j] = K.sqrt(K.square(y_true[i]-y_pred[i]) + K.square(y_true[i+1]-y_pred[i+1]))`
`        j = j + 1`
`        i= i + 2`
`    return K.means(ktotal)`
It still have error, I know maybe because of tensor we must use backend operation, but I dont know what to do.
2) How can we create new metrics, for example we define our custom metric of accuracy? like in my case `acc = 1-(max distance/mean(euclidean)`
3) How we monitor using new metrics and save the best weight using our custom metrics?
Any suggestion is welcomed. Thank you.

**Comment by Danielhiversen** [comment]
Here you find an example of how a custom dice function is as loss function:
https://github.com/EdwardTyantov/ultrasound-nerve-segmentation/search?utf8=%E2%9C%93&q=dice_coef&type=

**Comment by herleeyandi** [comment]
@Danielhiversen Yes I have check it before post this question. From that post I know that it must using Keras Backend. But I just ask how about we need iterate through index?, because in my case I need taking it by indexevery 2 elements like `d1 = sqrt((predx1-truex1)^2+(predy1-truey1)^2)` then compute `d2, d3, d4`. For example if I just take one by one not per 2 elements in array, I think I just simply can do like in that example.
